### PR TITLE
refactor(comp:space): use tsx

### PR DESCRIPTION
**Pull request by danranVm**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [our guidelines](https://github.com/IduxFE/idux/blob/main/packages/site/src/docs/Contributing.zh.md#commit)
- [x] Tests for the changes have been added/updated or not needed
- [x] Docs and demo have been added/updated or not needed

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Component style update
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Application (the showcase website) / infrastructure changes
- [ ] Other... Please describe:

### CODE REVIEW

File: packages/cdk/utils/src/vNode.ts [hunk 1]
```diff
@@ -1,14 +1,16 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (LaamGinghong) [inline packages/cdk/utils/src/vNode.ts:1]: reason

File: packages/cdk/utils/src/vNode.ts [hunk 2]
```diff
@@ -74,3 +76,31 @@ export function hasSlot(slots: Slots, key = 'default', ...options: unknown[]): b
   const firstValidNode = getFirstValidNode(vNodes)
   return isValidElementNode(firstValidNode)
 }
+
+export function isEmptyElement(node: VNodeChild): boolean {
+  return (
+    isComment(node) ||
+    (isFragment(node) && (node as any).children.length === 0) ||
+    (isText(node) && (node as any).children.trim() === '')
```
Comment (LaamGinghong) [inline packages/cdk/utils/src/vNode.ts:84]: 应该可以做严格的类型校验

File: packages/components/space/src/space.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+import { useGlobalConfig } from '@idux/components/config'
+import { spacePropsDef } from './types'
+
+export default defineComponent({
+  name: 'IxSpace',
+  props: spacePropsDef,
+  setup(props) {
+    const config = useGlobalConfig('space')
+    const classes = useClasses(props)
+    const size$$ = computed(() => props.size ?? config.size)
```
Comment (LaamGinghong) [inline packages/components/space/src/space.tsx:16]: 这种命名规范可以讨论一下，或者不使用计算，放到render里面

File: packages/cdk/utils/src/vNode.ts [hunk 4]
```diff
@@ -74,3 +76,31 @@ export function hasSlot(slots: Slots, key = 'default', ...options: unknown[]): b
...
+      result.push(...node)
+    } else if (isFragment(node)) {
+      result.push(...(node as any).children)
+    } else {
+      result.push(node)
+    }
+  })
+
+  return result.filter(c => !isEmptyElement(c))
+}
```
Comment (LaamGinghong) [inline packages/cdk/utils/src/vNode.ts:106]: 这个文件覆盖率好像有点低啊

**Review by LaamGinghong — CHANGES_REQUESTED** [review]

File: packages/cdk/utils/src/vNode.ts (hunk 2 above)
Comment (danranVm) [reply packages/cdk/utils/src/vNode.ts:84]: 做不了， vue 没有导出相关的类型。

File: packages/cdk/utils/src/vNode.ts (hunk 4 above)
Comment (danranVm) [reply packages/cdk/utils/src/vNode.ts:106]: 新增的几个函数没加单测，后面补吧。

File: packages/components/space/src/space.tsx (hunk 3 above)
Comment (danranVm) [reply packages/components/space/src/space.tsx:16]: 之前就讨论过的，就是这个命名。后面跟两个 $$

**Pull request merged**